tests/tests_geomstats/test_hypersphere.py

```diff
@@ -1,911 +1,701 @@
-        geodesic = self.metric.geodesic(
-            initial_point=initial_point[:2], end_point=initial_point[2:]
-        )
-        t = gs.linspace(start=0.0, stop=1.0, num=n_geodesic_points)
-        points = geodesic(t)
-        result = points[:, -1]
-        expected = initial_point[2:]
-        self.assertAllClose(expected, result)
-
-    def test_inner_product(self):
-        tangent_vec_a = gs.array([1.0, 0.0, 0.0, 0.0, 0.0])
-        tangent_vec_b = gs.array([0.0, 1.0, 0.0, 0.0, 0.0])
-        base_point = gs.array([0.0, 0.0, 0.0, 0.0, 1.0])
-        result = self.metric.inner_product(tangent_vec_a, tangent_vec_b, base_point)
-        expected = 0.0
-
-        self.assertAllClose(expected, result)
-
-    def test_inner_product_vectorization_single_samples(self):
-        tangent_vec_a = gs.array([1.0, 0.0, 0.0, 0.0, 0.0])
-        tangent_vec_b = gs.array([0.0, 1.0, 0.0, 0.0, 0.0])
-        base_point = gs.array([0.0, 0.0, 0.0, 0.0, 1.0])
-
-        result = self.metric.inner_product(tangent_vec_a, tangent_vec_b, base_point)
-        expected = 0.0
-        self.assertAllClose(expected, result)
-
-        tangent_vec_a = gs.array([[1.0, 0.0, 0.0, 0.0, 0.0]])
-        tangent_vec_b = gs.array([0.0, 1.0, 0.0, 0.0, 0.0])
-        base_point = gs.array([0.0, 0.0, 0.0, 0.0, 1.0])
-
-        result = self.metric.inner_product(tangent_vec_a, tangent_vec_b, base_point)
-        expected = gs.array([0.0])
-        self.assertAllClose(expected, result)
-
-        tangent_vec_a = gs.array([1.0, 0.0, 0.0, 0.0, 0.0])
-        tangent_vec_b = gs.array([[0.0, 1.0, 0.0, 0.0, 0.0]])
-        base_point = gs.array([0.0, 0.0, 0.0, 0.0, 1.0])
-
-        result = self.metric.inner_product(tangent_vec_a, tangent_vec_b, base_point)
-        expected = gs.array([0.0])
-        self.assertAllClose(expected, result)
-
-        tangent_vec_a = gs.array([[1.0, 0.0, 0.0, 0.0, 0.0]])
-        tangent_vec_b = gs.array([[0.0, 1.0, 0.0, 0.0, 0.0]])
-        base_point = gs.array([0.0, 0.0, 0.0, 0.0, 1.0])
-
-        result = self.metric.inner_product(tangent_vec_a, tangent_vec_b, base_point)
-        expected = gs.array([0.0])
-        self.assertAllClose(expected, result)
-
-        tangent_vec_a = gs.array([[1.0, 0.0, 0.0, 0.0, 0.0]])
-        tangent_vec_b = gs.array([[0.0, 1.0, 0.0, 0.0, 0.0]])
-        base_point = gs.array([[0.0, 0.0, 0.0, 0.0, 1.0]])
-
-        result = self.metric.inner_product(tangent_vec_a, tangent_vec_b, base_point)
-        expected = gs.array([0.0])
-        self.assertAllClose(expected, result)
-
-    def test_diameter(self):
-        dim = 2
-        sphere = Hypersphere(dim)
-        point_a = gs.array([[0.0, 0.0, 1.0]])
-        point_b = gs.array([[1.0, 0.0, 0.0]])
-        point_c = gs.array([[0.0, 0.0, -1.0]])
-        result = sphere.metric.diameter(gs.vstack((point_a, point_b, point_c)))
-        expected = gs.pi
-        self.assertAllClose(expected, result)
-
-    def test_closest_neighbor_index(self):
```

Comment:
missing?

Reply:
this is kind of useless. we are actually indexing into points and checking again point belongs to points ?